File: view/src/main/java/com/chen/view/MainActivity.java
package com.chen.view;

import androidx.appcompat.app.AppCompatActivity;

import android.animation.ObjectAnimator;
import android.os.Bundle;
import android.view.View;
import android.view.animation.AnimationUtils;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //View v = findViewById(R.id.v);
        //ObjectAnimator.ofFloat(v, "translationX", 0, 300).setDuration(1000).start();
    }
}

File: architecture/src/main/java/com/chen/architecture/MainActivity.kt
package com.chen.architecture

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.fragment.app.Fragment
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initView()
    }

    private fun initView() {
        changeFragment(NormalFragment.newInstance())
        btnNormal.setOnClickListener { changeFragment(NormalFragment.newInstance()) }
        btnMVC.setOnClickListener { changeFragment(MVCFragment.newInstance()) }
        btnMVP.setOnClickListener { changeFragment(MVPFragment.newInstance()) }
        btnMVVM.setOnClickListener { changeFragment(MVVMFragment.newInstance()) }
    }

    private fun changeFragment(fragment: Fragment) {
        supportFragmentManager.beginTransaction().replace(R.id.flContent, fragment).commit()
    }
}
